docs/reference/measure_writing_guide.md

```diff
@@ -1694,6 +1741,134 @@ Below is an example run method that manipulates and reports values related to a
     return true
   end</code></pre>
 
+## Using files and using ExternalFile in a measure
+
+It is sometimes needed to create support files in the process of running the measure. This section describes how path handling works in the context of a measure.
+
+When running a measure, the current working directory is something like './run/000_measure_class_name/' (this is the output that `File.realpath('./')` will give you).
+None of the files created in this directory will actually be copied over, unless it is the special case described in the section 'Reporting Measures' > 'Outputting an HTML (or other file)'.
+
+### Measure resource files
+
+#### Where to place them
+Resource files for a measure should be placed in the `resources/` subfolder like the Measure File Structure section indicates to do with additional ruby code.
+Note that nested levels are not accepted, meaning that given the below tree, anything in `resources/subfolder` is not valid: `anotherschedulefile.csv` will not be copied over with the measure.
+
+```
+├── measure.rb
+├── measure.xml
+├── resources
+│   ├── schedulefile.csv
+    │   ├── subfolder
+    │   │   ├── anotherschedulefile.csv
+```
+
+#### How to access them inside a measure
+
+You can locate your measure resources by using a relative path to the `measure.rb` you are running by using `File.join(File.dirname(__FILE__), 'schedulefile.csv')`
+
+#### How to use an ExternalFile inside a measure
+
+The constructor for `ExternalFile` will automatically copy the file at the path you provide to the first element in `WorkflowJSON::filePaths[0]`. When running a measure, the openstudio-workflow gem prepends the `generated_files` subdirectory.
+
+```ruby
+class CreateScheduleFile < OpenStudio::Measure::ModelMeasure
+
+  [...]
+
+  def run(model, runner, user_arguments)
+
+    # Locate the resource file (this resolves to something like './measures/resources/schedulefile.csv')
+    csv_in_path = File.join(File.dirname(__FILE__), 'schedulefile.csv')
+
+    # Instantiate an External File: this will automatically copy to first path in WorkflowJSON: `runner.workflow.filePaths[0]`, typically './generated_files/'
+    externalFile_ = OpenStudio::Model::ExternalFile::getExternalFile(model, csv_in_path)
+    if (!externalFile_)
+      runner.registerError("Failed to instantiate an External File")
+      return false
+    end
+
+    column = 1
+    rowsToSkip = 1
+    scheduleFile = OpenStudio::Model::ScheduleFile.new(externalFile_.get(), column, rowsToSkip)
+    scheduleFile.setName("ExampleScheduleFile")
+
+    return true
+end
+```
```

Comment:
This section is fair game and general enough. I definitely think it should be added.